fl4health/reporting/json_reporter.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+import json
+import uuid
+from logging import INFO
+from pathlib import Path
+from typing import Any
+
+from flwr.common.logger import log
+
+from fl4health.reporting.base_reporter import BaseReporter
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    """
+    Converts a datetime object to string in order to make json encoding easier.
+    """
+
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, datetime.datetime):
+            return str(obj)
+        else:
+            return json.JSONEncoder.default(self, obj)
+
+
+class FileReporter(BaseReporter):
+    def __init__(
+        self,
+        run_id: str | None = None,
+        output_folder: str | Path = Path("metrics"),
+    ):
+        """Reports data each round and saves as a json.
+
+        Args:
+            run_id (str | None, optional): the identifier for the run which these
+                metrics are from. If left as None will check if an id is provided during
+                initialize, otherwise uses a UUID.
+            output_folder (str | Path): the folder to save the metrics to. The metrics
+                will be saved in a file named {output_folder}/{run_id}.json. Optional,
+                default is "metrics".
+        """
+        self.run_id = run_id
+
+        self.output_folder = Path(output_folder)
+        self.metrics: dict[str, Any] = {}
+
+        self.output_folder.mkdir(exist_ok=True)
+        assert self.output_folder.is_dir(), f"Output folder '{self.output_folder}' is not a valid directory."
+
+    def initialize(self, **kwargs: Any) -> None:
```

Comment:
We are at least assuming kwargs is a dict with string keys the way we are using it. If we imagine there is any use case where this would not be a dict with string keys, we can keep it as any otherwise maybe we do dict[str, Any]

Reply:
Can't think of any cases now where we would want the keys to be something other than strings, was just trying to keep it general. But i'm happy to specify dict[str, Any] and let someone else change it if they need non-string keys